cpp/include/raft/util/cuda_dev_essentials.cuh

```diff
@@ -114,4 +116,19 @@ HDI void swapVals(T& a, T& b)
   b     = tmp;
 }
 
+/**
+ * @brief Convert half to float is needed
+ * @tparam T the datatype of the value
+ * @param a need to convert
+ */
+template <typename T>
+HDI auto half2float(T& a)
```

Comment:
I think this API should be renamed - since this not only converts from half2float, but also handles conversions when T=float.

I also think we probably want to have both an input and output template parameters for the conversion - rather than have a template parameter for the input, and assume the output is always a float. 

The faiss codebase has a generic 'convert' operator that does this - 
(https://github.com/facebookresearch/faiss/blob/dd72e4121dc6684c6fbf289949ba4526a54d9f3b/faiss/gpu/utils/ConversionOperators.cuh#L39-L45) which seems to work pretty well